x-pack/plugins/cases/public/components/all_cases/index.tsx

```diff
@@ -422,7 +429,17 @@ export const AllCases = React.memo<AllCasesProps>(
 
     const tableRowProps = useCallback(
       (theCase: Case) => {
-        const onTableRowClick = memoize(() => {
+        const onTableRowClick = memoize(async () => {
+          if (alertData != null) {
+            await postComment({
```

Comment:
this was previously in `security_solution/public/cases/components/timeline_actions/add_to_case_action.tsx`

Reply:
Imo, this shouldn't be inside the component. It is better to be implemented inside the `onRowClick` where the consumer of the component pass it as a prop. If security solution, needs access to the `postComment` function we can provided it through our `casesClient` (backend).

Reply:
i didnt want the ui components to rely on any api methods since actions dont, they should just work

Reply:
I see what you mean. I talk about it with @XavierM and we should break the AllCases components into smaller components and remove the `isModal` logic. This way we can move the `onTableRowClick` and `alertData` logic inside a new modal called, for example, `AttachToCaseModal` which can be exported through the `CasesUiClient`.

Example:

```
// AllCases component
interface Props {
  configureCasesHref: string;
  createCaseHref: string;
  disabledStatuses?: CaseStatuses[];
  getCaseDetailsHref: (caseDetails: CaseDetailsHrefSchema) => string;
  onCaseDetailsNavClick: (caseDetails: CaseDetailsHrefSchema) => void;
  onConfigureCasesNavClick?: (ev: React.MouseEvent) => void;
  onCreateCaseNavClick?: (ev: React.MouseEvent) => void;
  userCanCrud: boolean;
}

<>
  <AllCasesHeader>
  <AllCasesFilters>
  <AllCasesTable>
</>

// AttachToCase modal
interface Props {
   onAttachment: () => void;
}

<>
  <AllCasesFilters>
  <AllCasesTable>
</>

```